src/containers/widget/index.js

```diff
@@ -1,35 +1,64 @@
 import { h } from "preact";
-import { useState } from "preact/hooks";
 import { CacheProvider } from "@emotion/react";
+import { useState, useEffect } from "preact/hooks";
 
 import Modal from "../../components/modal";
 import Button from "../../components/button";
 
+import InvoiceService from "../../services/invoice";
+
 import DonateScreen from "../../views/donate-screen";
 import WelcomeScreen from "../../views/welcome-screen";
 
 import { StyledCache } from "../../helpers/styled-cache";
 import { WidgetWrapper } from "../../helpers/widget-wrapper";
 
+const API_TOKEN = "L7bRwjnMTvKpLgRhzScqi5Y8";
+const DEFAULT_API_BASE_URL = "https://area402.herokuapp.com";
+
 const Widget = ({
+  currency,
   imageSrc,
   showModal,
+  apiBaseUrl,
   screenName,
   widgetTitle,
   welcomeMessage,
+  paymentOptions,
 }) => {
   const [selectedAmount, setSelectedAmount] = useState(0);
   const [isModalOpen, setModalIsOpen] = useState(showModal);
   const [currentScreen, setCurrentScreen] = useState(screenName);
 
+  useEffect(() => {
+    console.log("selected amount", selectedAmount);
+  }, [selectedAmount]);
+
   const openModal = () => setModalIsOpen(true);
 
   const closeModal = () => setModalIsOpen(false);
 
-  const handleDonateNextClick = (value) => setSelectedAmount(value);
-
   const handleDonateClick = () => setCurrentScreen("donate-screen");
 
+  const paymentPagetRenderer = (invoiceDetails) => {
+    console.log("invoiceDetails", invoiceDetails);
+    // setCurrentScreen("paymeny-screen");
+  };
+
+  const handleDonateNextClick = async (value) => {
+    setSelectedAmount(value);
+
+    const invoiceOptions = {
+      value,
+      apiToken: API_TOKEN,
+      baseURL: apiBaseUrl,
+      paymentRequestRenderer: paymentPagetRenderer,
+    };
+    const invoiceService = new InvoiceService(invoiceOptions);
+
+    invoiceService.requestPayment();
```

Comment:
please not this currently returns a promise which resolves when the payment is received. - then we should display the success screen.

Reply:
line 19/20 in the invoice service: https://github.com/bumi/area402-widget/pull/29/files#diff-11dec6989c54070d1f26940347dee3199af4b02aad997a4724b724c392dab984R19

Reply:
yeah, I am aware of that I know the code, there's no callback that comes to this line here, it more like the whole functionality is passed down to the `paymentRequestRenderer` callback function.

Reply:
> line 19/20 in the invoice service: https://github.com/bumi/area402-widget/pull/29/files#diff-11dec6989c54070d1f26940347dee3199af4b02aad997a4724b724c392dab984R19

Yes, I am aware of how the Invoice is implemented. Understood after going through the code.

Reply:
we can change the implementation of the promise or what `requestPayment` does if needed and if that works better with the preact flow.